Sources/CSV/CSVWriter.swift

```diff
@@ -183,8 +183,10 @@ extension CSVWriter {
         var value = value
 
         var quoted = quoted
-        if value.contains("\"") || value.contains(",") {
-            quoted = true
+        if !quoted {
+            if value.contains("\"") || value.contains(configuration.delimiter) || value.contains("\r") || value.contains("\n") {
```

Comment:
RFC 4180 says the following (I forgot the newlines...):

```
   6.  Fields containing line breaks (CRLF), double quotes, and commas
       should be enclosed in double-quotes.  For example:

       "aaa","b CRLF
       bb","ccc" CRLF
       zzz,yyy,xxx
```

----

https://www.ietf.org/rfc/rfc4180.txt